platform/windows/display_server_windows.cpp

```diff
@@ -531,10 +531,31 @@ DisplayServer::ScreenOrientation DisplayServerWindows::screen_get_orientation(in
 }
 
 void DisplayServerWindows::screen_set_keep_on(bool p_enable) {
+	if (keep_screen_on == p_enable) {
+		return;
+	}
+
+	if (p_enable) {
+		const String reason = "Godot Engine running with display/window/energy_saving/keep_screen_on = true";
+
+		REASON_CONTEXT context;
+		context.Version = POWER_REQUEST_CONTEXT_VERSION;
+		context.Flags = POWER_REQUEST_CONTEXT_SIMPLE_STRING;
+		context.Reason.SimpleReasonString = (LPWSTR)(reason.utf16().ptrw());
+		power_request = PowerCreateRequest(&context);
+		PowerSetRequest(power_request, POWER_REQUEST_TYPE::PowerRequestSystemRequired);
```

Comment:
PowerSetRequest() and PowerClearRequest() can fail too, but in those cases we probably can't do much, except maybe bail and not update keep_screen_on state. Should still close the hande if that succeeded, though.